src/main/java/danchat/ui/DanChat.java

```diff
@@ -0,0 +1,346 @@
+            e.printStackTrace();
+        }
+    }
+
+    private static void processUserInput(String userInput) {
+        String[] splitInput = splitCommandAndDetail(userInput);
+        String command = splitInput[0];
+        String detail = splitInput[1];
+
+        executeUserCommandAndDetail(command, detail);
+    }
+
+    private static String[] splitCommandAndDetail(String userInput) {
+        String[] commandAndDetail = userInput.split(" ", 2);
+        if (commandAndDetail.length == 2) {
+            return commandAndDetail;
+        }
+        return new String[] {commandAndDetail[0], null};
+    }
+
+    private static void executeUserCommandAndDetail(String command, String detail) {
+        try {
+            if (isValidByeCommand(command, detail)) {
+                executeByeCommand();
+            } else if (isValidListCommand(command, detail)) {
+//                executeListCommand(tasks);
+                executeListCommand(taskList);
+            } else if (isValidUnmarkCommand(command, detail)) {
+                executeUnmarkCommand(detail);
+                saveChangeToFile();
+            } else if (isValidMarkCommand(command, detail)) {
+                executeMarkCommand(detail);
+                saveChangeToFile();
+            } else if (isValidTodoCommand(command, detail)) {
+                executeTodoCommand(detail);
+                saveChangeToFile();
+            } else if (isValidDeadlineCommand(command, detail)) {
+                executeDeadlineCommand(detail);
+                saveChangeToFile();
+            } else if (isValidEventCommand(command, detail)) {
+                executeEventCommand(detail);
+                saveChangeToFile();
+            } else if (isValidDeleteCommand(command, detail)) {
+                executeDeleteCommand(detail);
+                saveChangeToFile();
+            } else {
+                throw new IllegalCommandException("Unknown Command: " + command + " " + detail);
+            }
+        } catch (DanException e) {
+            System.out.println("Caught Exception: " + e.getMessage());
+        }
+    }
+
+    private static String getUserInput() {
+        String userInput = SCANNER.nextLine();
+        while (userInput.trim().isEmpty()) {
+            userInput = SCANNER.nextLine();
+        }
+        return userInput;
+    }
+
+    private static void executeEventCommand(String detail) throws MissingDateException {
+        String[] splitDetails = detail.split(EVENT_BEGIN_DATE_SEPERATOR, 2);
+        if (splitDetails.length < 2) {
+            throw new MissingDateException(ERROR_MISSING_BEGIN_DATE);
+        }
+        String description = splitDetails[0].trim();
+
+        String duration = splitDetails[1].trim();
+        String[] splitDurations = duration.split(EVENT_END_DATE_SEPERATOR, 2);
+        if (splitDurations.length < 2) {
+            throw new MissingDateException(ERROR_MISSING_END_DATE);
+        }
+        String from = splitDurations[0].trim();
```

Comment:
There are many magic literals here it would be helpful to refactor them or add some concise comments on what is happening here.